src/self-managed-debug/src/main.rs

```diff
@@ -0,0 +1,722 @@
+    };
+
+    fn format_network_policy_peer(peers: &[NetworkPolicyPeer]) -> String {
+        peers
+            .iter()
+            .filter_map(|peer| {
+                peer.ip_block.as_ref().map(|ip_block| {
+                    if let Some(except) = &ip_block.except {
+                        format!("{}[{}]", ip_block.cidr, except.join(","))
+                    } else {
+                        ip_block.cidr.clone()
+                    }
+                })
+            })
+            .collect::<Vec<_>>()
+            .join(" ")
+    }
+
+    // // NetworkPolicies
+    let netpol_info: Vec<NetworkPolicyInfo> = network_policies
+        .into_iter()
+        .filter_map(|netpol| {
+            netpol.metadata.name.clone().map(|name| {
+                let spec = netpol.spec.unwrap_or_default();
+
+                let ing_ports = spec.ingress.as_ref().map(|ingress| {
+                    ingress
+                        .iter()
+                        .filter_map(|rule| rule.ports.as_ref().map(format_network_policy_ports))
+                        .collect::<Vec<_>>()
+                        .join(",")
+                });
+
+                let ing_block = spec.ingress.as_ref().map(|ingress| {
+                    ingress
+                        .iter()
+                        .filter_map(|rule| {
+                            rule.from
+                                .as_ref()
+                                .map(|from| format_network_policy_peer(from))
+                        })
+                        .collect::<Vec<_>>()
+                        .join(",")
+                });
+
+                let egress_ports = spec.egress.as_ref().map(|egress| {
+                    egress
+                        .iter()
+                        .filter_map(|rule| rule.ports.as_ref().map(format_network_policy_ports))
+                        .collect::<Vec<_>>()
+                        .join(",")
+                });
+
+                let egress_block = spec.egress.as_ref().map(|egress| {
+                    egress
+                        .iter()
+                        .filter_map(|rule| {
+                            rule.to.as_ref().map(|to| format_network_policy_peer(to))
+                        })
+                        .collect::<Vec<_>>()
+                        .join(",")
+                });
+
+                NetworkPolicyInfo {
+                    name,
+                    ing_ports: ing_ports.unwrap_or_default(),
+                    ing_block: ing_block.unwrap_or_default(),
+                    egress_ports: egress_ports.unwrap_or_default(),
+                    egress_block: egress_block.unwrap_or_default(),
+                    age: format_age(&netpol.metadata),
+                }
+            })
+        })
+        .collect();
+    if !netpol_info.is_empty() {
+        writeln!(
+            file,
+            "NetworkPolicies:\n{}\n",
+            Table::new(netpol_info).with(Style::psql())
```

Comment:
Did you consider writing json (or something else structured) instead of formatting as a table? If it's json, we could always format it later, but the table doesn't allow us easily read the contents in a program.